Source/Core/Core/HW/GCMemcard/GCMemcardDirectory.cpp

```diff
@@ -32,96 +32,65 @@
 const int NO_INDEX = -1;
 static const char* MC_HDR = "MC_SYSTEM_AREA";
 
-int GCMemcardDirectory::LoadGCI(const std::string& file_name, bool current_game_only)
+bool GCMemcardDirectory::LoadGCI(GCIFile gci)
 {
-  File::IOFile gci_file(file_name, "rb");
-  if (gci_file)
+  // check if any already loaded file has the same internal name as the new file
+  for (const GCIFile& alreadyLoadedGci : m_saves)
```

Comment:
To match the remainder of the file
```suggestion
  for (const GCIFile& already_loaded_gci : m_saves)
```